pkg/api/v1/atlasproject_types.go

```diff
@@ -20,27 +20,47 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+// Important:
+// The procedure working with this file:
+// 1. Edit the file
+// 1. Run "make generate" to regenerate code
+// 2. Run "make manifests" to regenerate the CRD
 
-// AtlasProjectSpec defines the desired state of AtlasProject
+// Dev note: this file should be placed in "v1" package (not the nested one) as 'make manifests' doesn't generate the proper
+// CRD - this may be addressed later as having a subpackage may get a much nicer code
+
+func init() {
+	SchemeBuilder.Register(&AtlasProject{}, &AtlasProjectList{})
+}
+
+// AtlasProjectSpec defines the desired state of Project in Atlas
 type AtlasProjectSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
 
-	// Foo is an example field of AtlasProject. Edit AtlasProject_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// Name is the name of the Project that is created in Atlas by the Operator if it doesn't exist yet.
+	Name string `json:"name"`
+
+	// ConnectionSecret is the name of the Kubernetes Secret which contains the information about the way to connect to
+	// Atlas (organization ID, API keys). The default Operator connection configuration will be used if not provided.
+	// +optional
+	ConnectionSecret *SecretRef `json:"connectionSecretRef,omitempty"`
+
+	// ProjectIPAccessList allows to enable the API Access List for the Project. See more information at
+	// https://docs.atlas.mongodb.com/configure-api-access/
+	// +optional
+	ProjectIPAccessList []ProjectIPAccessList `json:"ipAccessList,omitempty"`
```

Comment:
[nit/q] for me `ProjectIPAccessList` has to do with https://docs.atlas.mongodb.com/reference/api/access-lists/ but your doc for this field makes me think this is for
https://docs.atlas.mongodb.com/reference/api/api-access-list/get-all-api-access-entries/
can you please clarify? if the field is for the project ip access list let's fix the link, if this is for API access list then could we rename the field to `APIAccessList`?

Reply:
reading the the details of `ProjectIPAccessList` I think this is a case of fixing the link unless you really meant API access and then implemented project access

Reply:
yes, you are right, thought that we could omit `project` prefix considering that this configuration is for the `AtlasProject` entity but I agree that this introduces confusion. Will change 👍